src/VisualStudio/VisualBasic/Impl/Options/StyleViewModel.vb

```diff
@@ -529,6 +529,123 @@ Class Customer2
 //]
 End Class"
 
+#Region "unused parameters"
+
+        Private Shared ReadOnly s_avoidUnusedParametersNever As String = $"
+Class C
+//[
+    Sub M(unused As Integer)
+    End Sub
+//]
+End Class
+"
+
+        Private Shared ReadOnly s_avoidUnusedParametersPrivateMethods As String = $"
+Class C1
+//[
+    ' {ServicesVSResources.Prefer_colon}
+    Private Sub M()
+    End Sub
+//]
+End Class
+
+Class C2
+//[
+    ' {ServicesVSResources.Over_colon}
+    Private Sub M(unused As Integer)
```

Comment:
note: the examples we show are pretty confusing IMO.  For example, in some code examples we show how calling somethig 'unused' makes things ok.  But for parameters this is shown as an example of something not ok that would then be removed.

Reply:
> calling somethig 'unused' makes things ok.

Which example are you referring to? The only example where unused things are shown ok or not flagged are the ones where you suppress this diagnostic or set Preference/Option to None. Its much clear in the UI then individual examples in code.

---
In reply to: [230606210](https://github.com/dotnet/roslyn/pull/30777#discussion_r230606210) [](ancestors = 230606210)

Reply:
hrmm... i may have misunderstood things then.  one of your examples has this:

```vb
        ' {ServicesVSResources.Prefer_colon}
        Dim unused = Computation()   ' Unused value is explicitly assigned to an unused local.
```

This is a 'preferred' form in one case.  So 'unused' is preferred here.  However, in the `Private Sub M(unused As Integer)` case, 'unused' is the bad thing.  That is the difference i'm calling out.  Thanks!

Reply:
Ah, correct. Let me clean it up to be more clear.

---
In reply to: [230988741](https://github.com/dotnet/roslyn/pull/30777#discussion_r230988741) [](ancestors = 230988741)